lib/chef/provider/package/rpm.rb

```diff
@@ -53,9 +53,11 @@ def load_current_resource
           @new_resource.version(nil)
 
           if @new_resource.source
-            unless ::File.exists?(@new_resource.source)
-              @package_source_exists = false
-              return
+            unless uri_scheme?(@new_resource.source)
```

Comment:
I would use `unless uri_scheme?(new_resource.source) || ::File.exists?(new_resource.source)` instead of nested unlesses